l2geth/core/rollup_fee.go

```diff
@@ -2,21 +2,38 @@ package core
 
 import (
 	"math/big"
+
+	"github.com/ethereum/go-ethereum/params"
 )
 
 // RollupBaseTxSize is the encoded rollup transaction's compressed size excluding
 // the variable length data.
 // Ref: https://github.com/ethereum-optimism/optimism/blob/91a9a3dcddf534ae1c906133b6d8e015a23c463b/packages/contracts/contracts/optimistic-ethereum/OVM/predeploys/OVM_SequencerEntrypoint.sol#L47
-const RollupBaseTxSize int = 96
+const RollupBaseTxSize uint64 = 96
 
 // CalculateFee calculates the fee that must be paid to the Rollup sequencer, taking into
 // account the cost of publishing data to L1.
-// Returns: (RollupBaseTxSize + len(data)) * dataPrice + executionPrice * gasUsed
+// Returns: (4 * zeroDataBytes + 16 * (nonZeroDataBytes + RollupBaseTxSize)) * dataPrice + executionPrice * gasUsed
```

Comment:
Confirming the implementation:

`4 * zeroDataBytes` is `zeroesCost`
`16 * (nonZeroDataBytes + RollupBaseTxSize)` is `onesCost`
Their sum `(4 * zeroDataBytes + 16 * (nonZeroDataBytes + RollupBaseTxSize))` is `dataCost`
`(4 * zeroDataBytes + 16 * (nonZeroDataBytes + RollupBaseTxSize)) * dataPrice` is `dataFee`
`executionPrice * gasUsed` is `executionFee`
And finally,
`(4 * zeroDataBytes + 16 * (nonZeroDataBytes + RollupBaseTxSize)) * dataPrice + executionPrice * gasUsed` is `fee`

The math looks good to me